src/connectors/impls/clickhouse.rs

```diff
@@ -0,0 +1,182 @@
+// Copyright 2021, The Tremor Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::connectors::prelude::*;
+
+use clickhouse_rs::{Block, Pool};
+
+#[derive(Default, Debug)]
+pub(crate) struct Builder {}
+
+#[async_trait::async_trait]
+impl ConnectorBuilder for Builder {
+    fn connector_type(&self) -> ConnectorType {
+        "clickhouse".into()
+    }
+
+    async fn build(&self, alias: &str, raw_config: &ConnectorConfig) -> Result<Box<dyn Connector>> {
+        // So something bad happens here and nobody know what or why.
+        let config = raw_config
+            .config
+            .as_ref()
+            .ok_or_else(|| ErrorKind::MissingConfiguration(alias.to_string()))?;
+
+        let config = ClickhouseConfig::new(config)?;
+
+        Ok(Box::new(Clickhouse { config }))
+    }
+}
+
+pub struct Clickhouse {
+    config: ClickhouseConfig,
+}
+
+#[async_trait::async_trait]
+impl Connector for Clickhouse {
+    async fn create_sink(
+        &mut self,
+        sink_context: SinkContext,
+        builder: SinkManagerBuilder,
+    ) -> Result<Option<SinkAddr>> {
+        let db_url = self.connection_url();
+        let sink = ClickhouseSink { db_url, pool: None };
+        builder.spawn(sink, sink_context).map(Some)
+    }
+
+    fn codec_requirements(&self) -> CodecReq {
+        CodecReq::Structured
+    }
+}
+
+impl Clickhouse {
+    fn connection_url(&self) -> String {
+        let ClickhouseConfig {
+            host,
+            port,
+            database,
+            ..
+        } = &self.config;
+        let port = port.unwrap_or(ClickhouseConfig::DEFAULT_PORT);
+
+        format!("tcp://{host}:{port}/{database}?compression=lz4")
```

Comment:
i am curious about the compression param. Is it required? Could we lift it to the connector configuration?